src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -75,6 +75,14 @@ public Uri AppConfigurationEndpoint
                     catch (FormatException) { }
                 }
 
+                // This is used only when a test passes in a ConfigurationClientManager but no endpoint or connection string is provided.
```

Comment:
I know this test-code-within-the-library pattern wasn't introduced in this PR, but I don't like that it's happening. I feel like we should be able to avoid it with a mocked dependency.

Reply:
Yea I agree it's weird, I think there might be a way that I just didn't see before so I'll try it out.

Reply:
I just changed the test to call `Connect` with an endpoint too but it isn't used, and it's stored in the endpoint value in the provider.